runtime/lua/vim/secure.lua

```diff
@@ -86,21 +113,75 @@ function M.read(path)
     trust[fullpath] = hash
   end
 
+  write_trust(trust)
+
+  return contents
+end
+
+--- Update the trust status of file at path in the trust database at
+--- $XDG_STATE_HOME/nvim/trust.
+---
+--- @param path (string) Path to a file to update status for.
+--- @param mode (string) One of the following:
+---   - 'allow': Add file to trust database and set it as trusted
+---   - 'deny': Add file to trust database and set it as denied
+---   - 'forget': Remove file from trust database
+--- @return (boolean, string|nil) success, errmsg: true and nil if the operation was successful,
+---   otherwise false and an error message.
+function M.trust(path, mode)
+  vim.validate({ path = { path, 's' } })
+  vim.validate({ allow = { mode, 's' } })
+  if mode ~= 'allow' and mode ~= 'deny' and mode ~= 'forget' then
+    return false, string.format('invalid mode: %s', mode)
+  end
+
+  local fullpath = vim.loop.fs_realpath(vim.fs.normalize(path))
+  if not fullpath then
+    return false, string.format('invalid path: %s', path)
+  end
+
+  local trust = read_trust()
+
+  if not trust[fullpath] and mode == 'forget' then
+    -- File is not in trust database - nothing to do
+    return true, nil
+  end
+
+  if trust[fullpath] == '!' and mode == 'deny' then
+    -- File is already denied - nothing to do
+    return true, nil
+  end
+
+  local hash
   do
-    local f, err = io.open(vim.fn.stdpath('state') .. '/trust', 'w')
+    local f = io.open(fullpath, 'r')
     if not f then
-      error(err)
-    end
-
-    local t = {}
-    for p, h in pairs(trust) do
-      t[#t + 1] = string.format('%s %s\n', h, p)
+      return false, 'file does not exist: ' .. path
     end
-    f:write(table.concat(t))
+    local contents = f:read('*a')
     f:close()
+    hash = vim.fn.sha256(contents)
   end
 
-  return contents
+  if trust[fullpath] == hash and mode == 'allow' then
+    -- File already exists in trust database - nothing to do
+    return true, nil
+  end
+
+  if mode == 'allow' then
+    -- Allow
```

Comment:
Redundant comments

Reply:
resolved in 3e7837abc58078f82b69a551fc829bc2adfeedb7